src/qibolab/backends.py

```diff
@@ -50,6 +52,45 @@ def __init__(self, platform):
         }
         self.compiler = Compiler.default()
 
+    @property
+    def qubits(self) -> list[Union[str, int]]:
+        """Returns the qubits in the platform."""
+        return list(self.platform.qubits)
+
+    @property
+    def connectivity(self) -> list[tuple[Union[str, int], Union[str, int]]]:
+        """Returns the list of connected qubits."""
+        return list(self.platform.pairs)
+
+    @property
+    def natives(self) -> list[str]:
+        """Returns the list of native gates supported by the platform."""
+        compiler = Compiler.default()
+        natives = [g.__name__ for g in list(compiler.rules)]
+
+        if "CZ" in natives and "CNOT" in natives:
+            for pair in self.connectivity:
+                cz = gates.CZ(*pair)
+                cnot = gates.CNOT(*pair)
+
+                if not self._is_gate_calibrated(cz, compiler):  # pragma: no cover
+                    natives.remove("CZ")
+                    break
+
+                if not self._is_gate_calibrated(cnot, compiler):
+                    natives.remove("CNOT")
+                    break
```

Comment:
```suggestion
        for gate in ["CZ", "CNOT"]:
            if gate in natives and not any(
                self._is_gate_calibrated(getattr(gates, gate)(*pair), compiler)
                for pair in self.connectivity
            ):
                natives.remove(gate)
```

The idea is that, instead of requiring the gate to be supported by all pairs (which would be the correct condition for transpilation), we are looking whether "the chip supports that gate".

This is a much more lenient condition, which may lead to a transpiled circuit not being actually executable.
However, that would be most likely a problem with the calibration provided, since the information we're interested in is whether a certain chip should be operated with CZs or CNOTs. Since we do not support per-qubit/per-pair natives (at least not yet), the support should be uniform - if it's not, it may be because it has not been yet calibrated - in which case it's better to fail in the execution of the transpiled circuit, since you may at least be able to execute "some" circuits (for the calibrated pairs).

The correct condition (currently implemented) would require to provide at least dummy parameters, even when the gate is not yet available (which is possible, and maybe we'll do that - but I'd say not now, and with the proposed implementation it will be just a matter of switching `any()` with `all()`).

Reply:
Moreover, you could even lift the list of gates to be checked as a top-level constant (in the proposed implementation - in the old one instead you would have to refactor anyhow to make it possible).

Reply:
The purpose is to find a gate that can be supported by the chip (all qubit pairs). Should it be `any(not ...)` instead of `not any(...)`?

Reply:
That's what I tried to explain above: `any(not ...)` is equivalent to `not all(...)`.

- the strict condition, `not all(...)`: every transpiled circuit can be run - rules out the entire two-qubit gates kind if missing even for a single pair
- the lenient condition, `not any(...)`: some transpiled circuits can not be run - it will also transpile circuits with two-qubit gates that can be run, but they would have not been transpiled with the strict condition

I was proposing the *lenient* one for the time being, postponing the error to the compiler (the disadvantage is that you may get your error late - but you will get fewer errors).
When (and if) we'll enforce uniformity at the platform level (gates will be present on all pairs, even when not calibrated), we could move to the strict one.

Reply:
> Ttbomu, it's either one or the other, not both - so returning both would make it impossible to correctly any circuit that contains both of them (or decompose to both). So, they have to be filtered to achieve minimal usability.

So, the lenient one is different from this? It will return both CX and CZ.

Reply:
Nope, the common scenario is that a chip will either support CNOT or CZ, not both (as written above).

If a gate is unsupported by the chip, it will not appear on any pair (thus it will be discarded even by the lenient condition). Instead, a supported gate may only appear on some pairs, because not every pair gets calibrated at the same time (we're now struggling to have even a single pair calibrated on some chips).

Ex: single pair calibrated on a CZ chip
- lenient condition: CZ available, circuits targeting just that pair can run (the other not, even if they are generated by the transpiler)
- strict condition: nothing available, no two-qubit gate circuit can run (since no one will get transpiled)

Reply:
Ok. so that's an exception for the dummy backend. Both the "CZ" and "CNOT" gates will be returned in dummy.

Reply:
> Ok. so that's an exception for the dummy backend. Both the "CZ" and "CNOT" gates will be returned in dummy.

Ah, yes. But `dummy` can actually run everything (it's just generating random numbers, the only thing he cares about are the measurements and the expect shape).